### PR TITLE
Add rules_bison v0.2.2

### LINKED ISSUE #204: wanted: jmillikin/rules_{bison, m4, flex}
We are currently transitioning our (now deprecated) [`bazel-eomii-registry`](https://github.com/eomii/bazel-eomii-registry-deprecated) to become a fork of the `bazel-central-registry`. We want to upstream the modules that are not reliant on our [`rules_ll`](https://github.com/eomii/rules_ll) to this repo, since I believe they would be useful for more users.

The repos [`rules_m4`](https://github.com/jmillikin/rules_m4), [`rules_flex`](https://github.com/jmillikin/rules_flex) and [`rules_bison`](https://github.com/jmillikin/rules_m4) are essentially always used together, so I'm tracking progress for all three here. I intend to first add some proper testing that our original module files lacked. Then I'll submit the patched variants to this repo and then upstream the patches to the original repos.

The repos in question are for instance used by [carbon](https://github.com/carbon-language/carbon-lang) and will in general likely be used by Bazel projects implementing programming languages. We also have a module for Carbon itself (very experimental, [here](https://github.com/eomii/bazel-eomii-registry-deprecated/tree/main/modules/carbon)), but testing that will likely not be straightforward and Carbon is IMO too unstable to become a module for now.

### PR DESCRIPTION
**Pull request by jmillikin**
As requested in https://github.com/bazelbuild/bazel-central-registry/issues/204

### CODE REVIEW

File: modules/rules_bison/0.2.2/MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+testutil = use_extension(
+    "//bison/internal:testutil_ext.bzl",
+    "rules_bison_testutil_ext",
+    dev_dependency = True,
+)
+use_repo(testutil, "rules_bison_testutil")
+
+register_toolchains(
+    "@rules_bison_testutil//toolchains:all",
+    dev_dependency = True,
```
Comment (fmeum) [inline modules/rules_bison/0.2.2/MODULE.bazel:44]: This parameter requires Bazel 6.2.0 or higher. Could you add a `bazel_compatibility` specification to `module`? https://bazel.build/rules/lib/globals/module#module has the details.

**Review by fmeum — COMMENTED** [review]
Windows CI is failing, it looks like it fails to find runfiles. Does rules_bison require runfiles symlinks?

File: modules/rules_bison/0.2.2/MODULE.bazel (hunk 1 above)
Comment (jmillikin) [reply modules/rules_bison/0.2.2/MODULE.bazel:45]: Done.

**Comment by jmillikin** [comment]
> Windows CI is failing, it looks like it fails to find runfiles. Does rules_bison require runfiles symlinks?

It requires runfiles, though not necessarily symlinks: https://github.com/jmillikin/rules_bison/blob/v0.2.2/.bazelci/presubmit.yml#L51-L57

Updated this PR to set the `--enable_runfiles` flag unconditionally (since it should be a noop in Linux and macOS).

**Comment by fmeum** [comment]
Looks like that's made progress, but CI is still failing.

**Comment by jmillikin** [comment]
I'm not sure what's going on in the current Windows message, but I don't have a local Windows installation to debug with, and CI in https://github.com/jmillikin/rules_bison is green on Windows.

I'll drop that platform from the BCR presubmit for now.

**Review by fmeum — APPROVED** [review]

**Pull request merged**